scripts/zpool

```diff
@@ -47,28 +47,49 @@ This is a resource agent that manages ZFS zpools.
 EOF
 }
 
+zpool_status_log() {
+	# Log zpool status to splunk for post-mortem.  Run in it the background
+	# in case zpool status hangs.  Wait two seconds for it to log.
+	nohup sh -c 'zpool status | logger -t resource_agent_fail -p local7.debug' &
+	sleep 2
+}
+
 zpool_monitor() {
     local rc
 
     # exit immediately if configuration is not valid
     zpool_validate_all || exit $?
 
-    ocf_run -q -info zpool status $OCF_RESKEY_pool
-
-    # This example assumes the following exit code convention
-    # for zpool status:
-    # 0: imported
-    # any other: not imported
-    case "$?" in
-        0)
-            rc=$OCF_SUCCESS
-            ocf_log debug "ZFS pool $OCF_RESKEY_pool is imported"
-            ;;
-        *)
-            rc=$OCF_NOT_RUNNING
-            ocf_log debug "ZFS pool $OCF_RESKEY_pool is not imported"
-            ;;
-    esac
+    if [ -e /proc/spl/kstat/zfs/$OCF_RESKEY_pool/state ] ; then
+        # Best case, we can read the pool health locklessly from the proc file.
+        state=$(cat /proc/spl/kstat/zfs/$OCF_RESKEY_pool/state)
+
+        if [ "$state" == "SUSPENDED" ] || [ "$state" == "FAULTED" ] || [ "$state" == "UNAVAIL" ]
+		ocf_log debug "ZFS pool $OCF_RESKEY_pool is $state"
+		zpool_status_log
+
+		# There's no good OCF fatal error code for this situation.
+		# Choose OCF_ERR_ARGS since the pool's "configuration is
+		# not valid".
+		rc=$OCF_ERR_ARGS
+	else
+		rc=$OCF_SUCCESS
+	fi
+
+    elif [ -d /proc/spl/kstat/zfs/$OCF_RESKEY_pool ] ; then
+        # 2nd best case, we see a /proc entry for the pool so
+        # we at least know it's imported.
+	rc=$OCF_SUCCESS
+    else
+        # Worst case, run zpool status for a crude check.
```

Comment:
I don't think we need the **zpool status** fallback.  The kstat directory for a pool appears on cider, which is zfs-06.5.8, and that seems like enough backwards compatability to me.

Reply:
Ok, I'll remove it